src/main/java/mate/academy/webshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,171 @@
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String query = "SELECT * FROM shopping_carts";
+        List<ShoppingCart> cartList = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart cart = getCartFromResultSet(resultSet);
+                cart.setProducts(
+                        getCartProducts(cart.getId()));
+                cartList.add(cart);
+            }
+            return cartList;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get carts", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+        String query = "UPDATE shopping_carts "
+                + "SET user_id = ?"
+                + " WHERE cart_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, cart.getUserId());
+            statement.setLong(2, cart.getId());
+            statement.executeUpdate();
+            deleteCart(cart.getId());
+            putCartProducts(cart);
+            return cart;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update cart", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM shopping_carts WHERE cart_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteCart(id);
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new RuntimeException(String.format("Can't delete cart with id:%d", id), e);
+        }
+    }
+
+    public void deleteCart(Long cartId) {
+        String query = "DELETE FROM shopping_carts_products "
+                + "WHERE cart_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, cartId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete cart", e);
+        }
+    }
+
+    public List<Product> getCartProducts(Long cartId) {
+        String query = "SELECT products.* FROM shopping_carts_products "
+                + "JOIN products USING (product_id) WHERE cart_id = ?";
+        List<Product> productList = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            return getProducts(cartId, query, productList, connection);
```

Comment:
А чому ти передаєш об'єкт connection в параметри? + productList в тебе пустий. Навіщо його передавати і в методі модифікувати вхідний параметр? Чим менше вхідних параметрів в методі - тим краще. Це з книги Clean code
```suggestion
            return getProducts(cartId, query);
```

Reply:
Пофіксив.